File: ai/intents/before-llm/llmPromptPayloadBuilder.js
const MAX_CONTEXT_MESSAGES = 6;
const MAX_CONTEXT_CHARS = 700;
export const DEFAULT_ASSISTANT_TIME_ZONE = 'Asia/Jerusalem';

const normalizeHistoryRole = (role) => (role === 'assistant' ? 'assistant' : 'user');

export const sanitizeHistoryForPrompt = (history = []) => {
  if (!Array.isArray(history)) return [];
  return history
    .slice(-MAX_CONTEXT_MESSAGES)
    .map((item) => ({
      role: normalizeHistoryRole(item?.role),
      content: String(item?.content || '').trim().slice(0, MAX_CONTEXT_CHARS)
    }))
    .filter((item) => item.content);
};

export const getCurrentDateForPrompt = () => {
  const parts = new Intl.DateTimeFormat('en-US', {
    timeZone: DEFAULT_ASSISTANT_TIME_ZONE,
    year: 'numeric',
    month: '2-digit',
    day: '2-digit'
  }).formatToParts(new Date());

  const byType = Object.fromEntries(parts.map((part) => [part.type, part.value]));
  return `${byType.year}-${byType.month}-${byType.day}`;
};

export const buildUserPromptPayload = ({ userInput, history }) => ({
  currentDate: getCurrentDateForPrompt(),
  timeZone: DEFAULT_ASSISTANT_TIME_ZONE,
  currentUserMessage: String(userInput || '').trim(),
  recentConversation: sanitizeHistoryForPrompt(history)
});

File: ai/intents/llm_prompt_payload_builder.py
from datetime import datetime, timezone

MAX_CONTEXT_MESSAGES = 6
MAX_CONTEXT_CHARS = 700
DEFAULT_ASSISTANT_TIME_ZONE = "Asia/Jerusalem"


def get_current_date_for_prompt() -> str:
    try:
        import pytz
        tz = pytz.timezone(DEFAULT_ASSISTANT_TIME_ZONE)
        now = datetime.now(tz)
    except Exception:
        now = datetime.now(timezone.utc)
    return now.strftime("%Y-%m-%d")


def sanitize_history_for_prompt(history: list) -> list:
    if not isinstance(history, list):
        return []
    result = []
    for item in history[-MAX_CONTEXT_MESSAGES:]:
        role = item.get("role")
        role = "assistant" if role == "assistant" else "user"
        content = str(item.get("content") or "").strip()[:MAX_CONTEXT_CHARS]
        if content:
            result.append({"role": role, "content": content})
    return result


def build_user_prompt_payload(*, user_input: str, history: list) -> dict:
    return {
        "currentDate": get_current_date_for_prompt(),
        "timeZone": DEFAULT_ASSISTANT_TIME_ZONE,
        "currentUserMessage": str(user_input or "").strip(),
        "recentConversation": sanitize_history_for_prompt(history),
    }
